core/lib/types/src/commitment.rs

```diff
@@ -209,7 +209,7 @@ impl L1BatchWithMetadata {
                         .to_vec(),
                 ),
                 Token::Bytes(self.metadata.l2_l1_messages_compressed.clone()),
-                Token::Bytes(self.construct_pubdata()),
+                Token::Bytes(self.header.pubdata_input.clone().unwrap()),
```

Comment:
`self.header.pubdata_input` is `len || data` AFAIU, we don't need len here, so `self.header.pubdata_input[32..]` should be used